libsolidity/ast/Types.cpp

```diff
@@ -830,7 +830,9 @@ TypePointer RationalNumberType::binaryOperatorResult(Token::Value _operator, Typ
 		auto mobile = mobileType();
 		if (!mobile)
 			return TypePointer();
-		return mobile->binaryOperatorResult(_operator, _other);
```

Comment:
This will fail for `x < 1` / `1 < x`. I think the proper fix could be to just replace the if body here by
```
		auto commonType = Type::commonType(shared_from_this(), _other);
		if (!commonType)
			return TypePointer();
		return commonType->binaryOperatorResult(_operator, _other);
```

Reply:
We use `Type::commonType` for many other implementations of `binaryOperatorResult`.

Reply:
Okay, I put that in. Tests passed on my end.